backend/controller/controller.go

```diff
@@ -693,16 +695,53 @@ func (s *Service) GetModuleContext(ctx context.Context, req *connect.Request[ftl
 			return connect.NewError(connect.CodeInternal, fmt.Errorf("could not get databases: %w", err))
 		}
 
-		response := modulecontext.NewBuilder(name).AddConfigs(configs).AddSecrets(secrets).AddDatabases(databases).Build().ToProto()
+		checksum := configurationMapChecksum(configs)
+		checksum = (checksum * 115163) + configurationMapChecksum(secrets)
+		checksum = (checksum * 454213) + configurationDatabaseChecksum(databases)
```

Comment:
@jonathanj-square what are these `115163` and `454213` numbers?

Reply:
Agreed: just use sha256 for everything, there's no need to truncate it and switch to a manual 64-bit checksum. Pass around the `hash.Hash` from `sha256.New()` to generate the checksum, then just compare `checksum.Sum()` results with `bytes.Compare()`